src/utils/functions.ts

```diff
@@ -0,0 +1,28 @@
+import { NewObjPizza, NewObjSandwich, NewObjSoup } from './types';
+import { FormValues } from './yupSchema';
+
+export const createObject = (obj: FormValues) => {
```

Comment:
To powinno się nazywać buildPayload, bo tu nie tworzysz nowego obiektu tylko ustalasz jego strukturę.
Nawet tego let nie potrzebujesz. Można to rozpisać tak:

```js
const baseData={
dishName: ...,
inne wspolne
}
```
potem w kazdym if

return {...baseData, toCoSieRozni}